### PR TITLE
Document How to review a PR

### PR DESCRIPTION
**Pull request by redhug1**
### What
This is a first pass.
Others may wish to add other language specific points.

See: https://trello.com/c/aBNctZOw/5054-document-how-to-review-a-pr

This a first attempt, so any missing parts can be added in future 20% time stories.

### How to review

Does the document make sense ?

### Who can review

Anyone.

### CODE REVIEW

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### Prerequisites, Philosophy and Offering feedback
+
+When you are making an observation or giving helpful feedback for a specific piece of code (or a general overall comment), if what you are writing is over ~50 words - it may be better to have a direct call or discuss via Slack and then maybe a comment can be placed in the code as to why something is done a particular way.
+
+**Teamwork** - Reviewing other peoples work is an act of teamwork and teamwork can be challenging as it requires compromise.
+
+**Reviews** - Are an opportunity for learning - so don't just say 'do this' - consider explaining why, especially when its a best practice. Try and explain the value (maybe with links to articles), but be careful that the links you reference are balanced.
+
+**Kiss** - Keep It Simple and Straightforward. Does the body of work being reviewed lend itself to Maintenance ? Where applicable, are there sufficient comments explaining why and how something is being done.
```
Comment (franmoore05) [inline training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:14]: The uppercase M in "Maintenance" should be lowercase.

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+---
+And last but not least, **`RED Flags - Test Code Removed`**:
+
+If you see test code removed - thoroughly check into the reasons why - they may be valid but often indicate some problem.
+
+## General things to consider
+
+### Before diving into the Review
+
+Sometimes a PR will pass all checks on a local machine and then fail in CI for some reason and the author of the PR put it into slack before waiting to see that all tests in CI had passed.
```
Comment (franmoore05) [inline training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:27]: "put" should be "puts"

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+- Familiarise yourself with the Trello Ticket of the work you are reviewing.
+- Be humble.
+- Suggest ways to improve or simplify the code.
+- Don't just look at whats been done, but also look out for anything that may have been missed.
+---
+- Does the code work - tests pass ?
+- Are functions and variables named well ?
+- Are there suffucient tests ?
+- Are tests meaningfull ?
```
Comment (franmoore05) [inline training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:58]: There is a typo in "Meaningfull", should be one l.

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- Is test code coverage adequate ?
+- Can you understand the code ?
+- Any sensitive data being revealed ?
+- Does code meet house style ?
+- Is the code clean - free of commented out code / not needed code ?
+- Is there adequate commenting ?
+- Is input data validated ?
+- Might the code suffer speed issues (benefit from, for example: pagination) ?
+- Is the code modular, where applicable ?
+- Does code and test code address ALL sunny and rainy day paths (capture / process all errors) ?
```
Comment (franmoore05) [inline training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:68]: "ALL" should not be in uppercase.

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+## Responding to a Review
+
+Thank the reviewer.
+
+Suggestions received in a review are just suggestions, they are not an instruction.
+
+Its OK to push back if for example the effort outweighs the value, or the value has not been clearly explained.
+
+Ask for or Offer clarification, in a call if needs be.
```
Comment (franmoore05) [inline training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:119]: "Offer" should start with a lowercase o

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 1 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:14]: Done

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 2 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:27]: Done

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 3 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:58]: Done

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 4 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:68]: Done

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 5 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md:119]: Done

**Review by franmoore05 — APPROVED** [review]

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 1 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md]: done

File: training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md (hunk 2 above)
Comment (redhug1) [reply training/culture-and-process/PR_REVIEW_AND_RESPONSE_GUIDANCE.md]: done

**Pull request merged**